js/react/pages/wishCards.jsx

```diff
@@ -7,57 +9,181 @@ import MantineProviderWrapper from '../utils/mantineProviderWrapper.jsx';
 
 const cardsPerPage = 24;
 
-function WishCards({ wishCards, user }) {
+function WishCards({ wishCards, user, agencies }) {
 	const [isLoading, setIsLoading] = useState(true);
-	const [cards, setCards] = useState([]);
+	const [cardData, setCardData] = useState(wishCards);
+	const [searchTextInput, setSearchTextInput] = useState('');
+	const [searchQueryParams, setSearchQueryParams] = useState({
+		showDonated: 'yes',
+		ageGroups: [],
+		agencyFilter: '',
+		sortOrder: [],
+	});
 	const [numCardsToShow, setNumCardsToShow] = useState(cardsPerPage);
 
 	useEffect(() => {
-		setCards(
-			wishCards.map((wishCard) => {
-				let attributes = {};
-
-				if (!user?._id) {
-					attributes = {
-						href: '/login',
-					};
-				} else {
-					attributes = {
-						href: `/wishcards/donate/${wishCard._id}`,
-					};
-				}
-
-				return (
-					<div key={wishCard._id} className="m-3 col-12 col-md-5 col-lg-4 col-xxl-3">
-						<WishCard wishCard={wishCard} attributes={attributes} />
-					</div>
-				);
-			}),
-		);
-
 		setTimeout(() => {
 			setIsLoading(false);
 		}, 1000);
-	}, []);
+	}, [cardData]);
+
+	useEffect(() => {
+		fetchSearchResults();
+	}, [searchQueryParams]);
 
 	function handleLoadMore() {
 		setNumCardsToShow(numCardsToShow + cardsPerPage);
 	}
 
+	const mapwishCard = (wishCard) => {
+		return (
+			<div key={wishCard._id} className="m-3 col-12 col-md-5 col-lg-4 col-xxl-3">
+				<WishCard
+					wishCard={wishCard}
+					attributes={
+						!user?._id
+							? { href: '/login' }
+							: { href: `/wishcards/donate/${wishCard._id}` }
+					}
+				/>
+			</div>
+		);
+	};
+
+	const handleSearchInputChange = (event) => {
+		const target = event.target;
+		const { value } = target;
+		setSearchTextInput(value);
+	};
+
+	const handleSearchSubmit = async (event) => {
+		event.preventDefault();
+		await fetchSearchResults();
+	};
+
+	const fetchSearchResults = async () => {
+		const {
+			data: { wishcards },
+		} = await axios.post(
+			'/wishcards/search/',
+			{
+				wishitem: searchTextInput,
+				showDonatedCheck: searchQueryParams.showDonated,
+				younger: searchQueryParams.ageGroups.includes('younger') ? 'true' : null,
+				older: searchQueryParams.ageGroups.includes('older') ? 'true' : null,
+				recentlyAdded: searchQueryParams.sortOrder.includes('1') ? 1 : 0,
+				agencyFilter: searchQueryParams.agencyFilter
+					? agencies.find((agency) => agency.agencyName == searchQueryParams.agencyFilter)
+							._id
+					: null,
+			},
+			{
+				headers: {
+					'Content-Type': 'application/json',
+				},
+			},
```

Comment:
headers can be omitted if it's not a formdata object or you need to include some special headers for the request